### ISSUE TITLE
[Java][0.2.0]java.lang.NoClassDefFoundError: org/springrain/org (wrong name: org/springrain/Org)

### ISSUE THREAD

**Issue by springrain** [alipay/fury#1005]
**Describe the bug**
Serialize the Org class name exception, If the class name is not Org, it can be serialized normally

**Expected behavior**
Normal serialization

**To Reproduce**
```java
package org.springrain;

import java.io.Serializable;
import java.util.List;

public class Org implements Serializable {
    private static final long serialVersionUID = 1L;
    //constructor
    public Org() {
    }
    List<Org> children;

    public List<Org> getChildren() {
        return children;
    }
    public void setChildren(List<Org> children) {
        this.children = children;
    }

}
```
```java
package org.springrain;

import io.fury.Fury;
import io.fury.ThreadSafeFury;
import io.fury.config.CompatibleMode;
import io.fury.config.Language;

public class Main {
    public static void main(String[] args) {
          ThreadSafeFury fury = Fury.builder().withLanguage(Language.JAVA)
                // Allow to deserialize objects unknown types,more flexible but less secure.
                .requireClassRegistration(false)
                .withDeserializeUnexistedClass(true)
                .withCompatibleMode(CompatibleMode.COMPATIBLE)
                .withRefTracking(true)
                // .withAsyncCompilation(true)
                .buildThreadSafeFury();

          // If the class name is not Org, it can be serialized normally
          fury.serialize(new Org());
    }
}
```

**Screenshots**
![image](https://github.com/alipay/fury/assets/4935705/b6330af3-b218-4e94-9da5-4391957ec675)

**Environment (please complete the following information):**
```
openjdk 21 2023-09-19 LTS
OpenJDK Runtime Environment Corretto-21.0.0.35.1 (build 21+35-LTS)
OpenJDK 64-Bit Server VM Corretto-21.0.0.35.1 (build 21+35-LTS, mixed mode, sharing)
```

**Comment by chaokunyang** [comment]
Hi @springrain , thanks for report this bug. Ican reproduce it, adn it seems a bug in janino https://github.com/janino-compiler/janino/issues/165. I believe I fixed it in #1006, please try it out

**Comment by chaokunyang** [comment]
Hi, we've just released fury 0.2.1: https://github.com/alipay/fury/releases/tag/v0.2.1 .It should have fixed this issue. Could you try it out?
